### PR TITLE
feat(crosswalk): add option to slowdown at occluded crosswalks

### PR DESCRIPTION
**Pull request by maxime-clem**
## Description

Add a feature to slowdown when passing a crosswalk if it occluded.

Requires the launch PR: https://github.com/autowarefoundation/autoware_launch/pull/807

## Tests performed

Evaluator (TIER IV INTERNAL LINK): https://evaluation.tier4.jp/evaluation/reports/7e2f793c-c950-5a48-a185-b40f42001ecd?project_id=prd_jt

## Effects on system behavior

slowdown when passing an occluded crosswalk.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

- New Feature: Added the ability to slow down when passing an occluded crosswalk. This feature includes new parameters for handling occlusions and a safety slowdown speed. The code initializes these parameters from the ROS parameter server. The `CrosswalkModule` class now supports slowing down based on occlusions, and a new function `applySafetySlowDownSpeed` applies a safety slowdown speed to the path. Dependencies `grid_map_core`, `grid_map_ros`, `grid_map_utils`, `interpolation`, and `nav_msgs` were added to the package.

### CODE REVIEW

**Comment by shmpwk** [comment]
@maxime-clem 
Do you have a chart to describe this occluded crosswalk?

File: planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.hpp [hunk 1]
```diff
@@ -155,6 +155,18 @@ class CrosswalkModule : public SceneModuleInterface
...
+    double occlusion_occluded_object_velocity;
+    float occlusion_slow_down_velocity;
+    double occlusion_time_buffer;
+    double occlusion_min_size;
+    int occlusion_free_space_max;
+    int occlusion_occupied_min;
+    bool occlusion_ignore_with_red_traffic_light;
+    bool occlusion_ignore_behind_predicted_objects;
+    std::vector<double> occlusion_ignore_velocity_thresholds;
+    double occlusion_extra_objects_size;
```
[the remark this answers is not in the document]
Comment (shmpwk) [reply planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.hpp:169]: ok

File: planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.hpp [hunk 2]
```diff
@@ -316,7 +328,8 @@
 private:
   // main functions
   void applySafetySlowDownSpeed(
-    PathWithLaneId & output, const std::vector<geometry_msgs::msg::Point> & path_intersects);
+    PathWithLaneId & output, const std::vector<geometry_msgs::msg::Point> & path_intersects,
+    const float speed);
```
[the remark this answers is not in the document]
Comment (shmpwk) [reply planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.hpp:332]: ok

File: planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.hpp [hunk 3]
```diff
@@ -437,6 +450,10 @@
 
   // whether use regulatory element
   const bool use_regulatory_element_;
+
+  // occluded space time buffer
+  std::optional<rclcpp::Time> current_initial_occlusion_time_;
+  std::optional<rclcpp::Time> most_recent_occlusion_time_;
```
[the remark this answers is not in the document]
Comment (shmpwk) [reply planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.hpp:456]: ok

**Comment by maxime-clem** [comment]
> @maxime-clem Do you have a chart to describe this occluded crosswalk?

Sorry I just added the documentation: https://autowarefoundation.github.io/autoware.universe/pr-6122/planning/behavior_velocity_crosswalk_module/#occlusion
Let me know if something is not clear.

**Review by takayuki5168 — APPROVED** [review]

**Pull request merged**